source/Calamari.Shared/Integration/Scripting/CombinedScriptEngine.cs

```diff
@@ -29,13 +29,15 @@ public CombinedScriptEngine(IEnumerable<IScriptWrapper> scriptWrapperHooks)
             this.scriptWrapperHooks = scriptWrapperHooks;
         }
 
+
         public ScriptSyntax[] GetSupportedTypes()
         {
-            return (CalamariEnvironment.IsRunningOnNix || CalamariEnvironment.IsRunningOnMac)
-                ? new[] { ScriptSyntax.Bash, ScriptSyntax.PowerShell, ScriptSyntax.CSharp, ScriptSyntax.FSharp, ScriptSyntax.Python }
-                : new[] { ScriptSyntax.PowerShell, ScriptSyntax.CSharp, ScriptSyntax.FSharp, ScriptSyntax.Python };
-        }
+            var preferredScriptSyntax = new [] { ScriptSyntaxHelper.GetPreferredScriptSyntaxForEnvironment() };
+            var commonScriptSyntax =  new[] { ScriptSyntax.PowerShell, ScriptSyntax.CSharp, ScriptSyntax.FSharp, ScriptSyntax.Python };
 
+            return preferredScriptSyntax.Concat(commonScriptSyntax).Distinct() as ScriptSyntax[];
```

Comment:
Don't cast. The `.Distinct()` returns an `IEnumerable<ScriptSyntax>`, yet the return type needs to be a `ScriptSyntax[]`. Remember that an `IEnumerable<ScriptSyntax>` is a very thin interface that just says "This thing can be enumerated" and nothing else. This is opposed to an array like `ScriptSyntax[]` which is a much more specific type that has a very specific way of storing its data in memory. An `IEnumerable<T>` might not even be in memory all at once, like an array is.

While an array implements `IEnumerable<T>`, it is not always true that an `IEnumerable<T>` is an array. In fact, in this case it as actually not true, so this "soft cast" to fail, resulting in a `null`. (Side note: Look up the difference between a soft and hard cast).

The correct way to do this is to just got `.Distinct().ToArray()`